google/cloud/aiplatform/jobs.py

```diff
@@ -1910,3 +1919,448 @@ def run(
+        )
+
+        api_client = empty_mdm_job.api_client
+
+        _LOGGER.log_create_with_lro(cls)
+
+        endpoint_resource_name = cls._get_endpoint_resource_name(endpoint)
+        gapic_mdm_job = (
+            gca_model_deployment_monitoring_job.ModelDeploymentMonitoringJob(
+                display_name=display_name,
+                endpoint=endpoint_resource_name,
+                model_deployment_monitoring_objective_configs=mdm_objective_config_seq,
+                logging_sampling_strategy=logging_sampling_strategy.as_proto(),
+                model_deployment_monitoring_schedule_config=schedule_config.as_proto(),
+                model_monitoring_alert_config=alert_config.as_proto(),
+                predict_instance_schema_uri=predict_instance_schema_uri,
+                analysis_instance_schema_uri=analysis_instance_schema_uri,
+                sample_predict_instance=sample_predict_instance,
+                stats_anomalies_base_directory=stats_anomalies_base_directory,
+                enable_monitoring_pipeline_logs=enable_monitoring_pipeline_logs,
+                labels=labels,
+                encryption_spec=encryption_spec_key_name,
+            )
+        )
+
+        gca_mdm_job = api_client.create_model_deployment_monitoring_job(
+            parent=parent,
+            model_deployment_monitoring_job=gapic_mdm_job,
+            timeout=timeout,
+        )
+        gca_mdm_job._gca_resource = gapic_mdm_job
+
+        empty_mdm_job._gca_resource = gca_mdm_job
+
+        mdm_job = empty_mdm_job
+
+        _LOGGER.log_create_complete(cls, mdm_job._gca_resource, "mdm")
+
+        _LOGGER.info(
+            "View Model Deployment Monitoring Job:\n%s" % mdm_job._dashboard_uri()
+        )
+
+        mdm_job._endpoint_resource_name = endpoint_resource_name
+
+        return mdm_job
+
+    def update(
+        self,
+        display_name: Optional[str] = None,
+        schedule_config: Optional[model_monitoring.ScheduleConfig] = None,
+        alert_config: Optional[model_monitoring.EmailAlertConfig] = None,
+        logging_sampling_strategy: Optional[model_monitoring.RandomSampleConfig] = None,
+        labels: Optional[Dict[str, str]] = None,
+        bigquery_tables_log_ttl: Optional[int] = None,
+        enable_monitoring_pipeline_logs: Optional[bool] = None,
+        objective_configs: Optional[
+            Union[
+                model_monitoring.EndpointObjectiveConfig,
+                Dict[str, model_monitoring.EndpointObjectiveConfig],
+            ]
+        ] = None,
+        deployed_model_ids: Optional[List[str]] = None,
+        training_dataset: Optional[str] = None,
+        training_prediction_skew_detection_config: Optional[
+            model_monitoring.EndpointSkewDetectionConfig
+        ] = None,
+        prediction_drift_detection_config: Optional[
+            model_monitoring.EndpointDriftDetectionConfig
+        ] = None,
+    ) -> "ModelDeploymentMonitoringJob":
```

Comment:
My apologies. This was copied over from the design doc and I forgot to change the actual return type hint. This function should return `None` since it is updating parameters of an already-existing MDM job. I have also reflected this in the return type hint.

Reply:
Originally I implemented this function to return `None` since it is technically a change in the job state and nothing needs to be returned. I did some more reading on method cascading in Python and decided that it's a better idea to return `self` for the `update`, `pause`, and `resume` functions since it will allow for method cascading.